src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -282,8 +282,10 @@ internal async Task ProcessMessage(Message receivedMessage, CancellationToken me
             {
                 try
                 {
+                    var nativeHeadersContainNServiceBusMessageIdHeader = false;
                     if (receivedMessage.MessageAttributes.TryGetValue(Headers.MessageId, out var messageIdAttribute))
                     {
+                        nativeHeadersContainNServiceBusMessageIdHeader = true;
```

Comment:
TODO: validate if checking for the presence of the message ID in the _native_ headers is enough to determine the sender is an NServiceBus endpoint. Are all versions of the transport setting the header?